core/src/main/java/org/apache/iceberg/PositionDeletesTable.java

```diff
@@ -130,18 +132,38 @@ private Schema calculateSchema() {
   public static class PositionDeletesBatchScan
       extends SnapshotScan<BatchScan, ScanTask, ScanTaskGroup<ScanTask>> implements BatchScan {
 
+    private boolean filterSet = false;
```

Comment:
This is messy, but the overall idea is, we either have a filter set on the metadata table, or the base table, and can only handle one of these.